lib/Arch/Sleigh/Arch.cpp

```diff
@@ -519,9 +538,56 @@ std::optional<int32_t> SingleInstructionSleighContext::oneInstruction(
 }
 
 
-OperandLifter::OpLifterPtr
-SleighArch::DefaultLifter(const remill::IntrinsicTable &intrinsics) const {
-  return std::make_unique<SleighLifter>(this, intrinsics);
+ContextUpdater::ContextUpdater(
+    DecodingContext curr_context_,
+    const std::unordered_map<std::string, std::string> &register_mapping_,
+    Sleigh &engine_)
+    : is_inter_procedural(false),
+      curr_context(curr_context_),
+      prev_context(std::move(curr_context_)),
+      register_mapping(register_mapping_),
+      engine(engine_) {}
+
+void ContextUpdater::dump(const Address &addr, OpCode opc, VarnodeData *outvar,
+                          VarnodeData *vars, int4 isize) {
+
+  if (opc == OpCode::CPUI_CALL || opc == OpCode::CPUI_CALLIND) {
+    this->is_inter_procedural = true;
+  }
+
+  // So we are updating a variable, if it's a target we either need to give it a new constant value or drop it to nonconstant
+  if (!outvar) {
+    return;
+  }
+
+  auto outvar_name =
+      this->engine.getRegisterName(outvar->space, outvar->offset, outvar->size);
+  auto target_remill_cont_reg = this->register_mapping.find(outvar_name);
+  if (target_remill_cont_reg == this->register_mapping.end()) {
+    return;
+  }
+
+  if (this->already_assigned.find(target_remill_cont_reg->second) !=
+          this->already_assigned.end() ||
+      !(opc == OpCode::CPUI_COPY && isVarnodeInConstantSpace(vars[0]))) {
+    LOG(INFO) << "Dropping " << target_remill_cont_reg->second;
```

Comment:
`DLOG` - sounds like something that should be printed when debugging rather normally.